src/test/java/StringManipulationTest.java

```diff
@@ -1,149 +1,164 @@
+        assertEquals("C", subStrings[1]);
+        assertEquals("D", subStrings[2]);
     }
+
     @Test
-    public void testGeSubStrings5() {
-        fail("Not yet implemented");
+    public void testGetSubStrings3() {
+        manipulatedString.setString("One Two"); //setting string
+        String[] subStrings = manipulatedString.getSubStrings(1, 2); //get substring array
+        assertEquals("One", subStrings[0]); //asserting elements
+        assertEquals("Two", subStrings[1]);
     }
+
     @Test
-    public void testGeSubStrings6() {
-        fail("Not yet implemented");
+    public void testGetSubStrings4() {
+        assertThrows(IllegalArgumentException.class, () -> manipulatedString.getSubStrings(3, 2)); //asserting exception for invalid arguments
     }
 
     @Test
-    public void testRestoreString1()
-    {
-        manipulatedstring.setString("art");
-        int [] array;
-        array=new int[]{1,0,2};
-        String restoreString = manipulatedstring.restoreString(array);
-        assertEquals(restoreString, "rat");
+    public void testGetSubStrings5() {
+        manipulatedString.setString("Short"); //setting string
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedString.getSubStrings(1, 2)); //asserting exception for out of bounds
     }
 
     @Test
-    public void testRestoreString2()
-    {
-        fail("Not yet implemented");
-
+    public void testGetSubStrings6() {
+        manipulatedString.setString("Just One"); //setting string
+        String[] subStrings = manipulatedString.getSubStrings(1, 1); //get substring array
+        assertEquals("Just", subStrings[0]); //asserting element
     }
 
     @Test
-    public void testRestoreString3()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString1() {
+        manipulatedString.setString("art"); //setting string
+        int[] array = {1, 0, 2}; //indices array
+        String restoreString = manipulatedString.restoreString(array); //restoring string
+        assertEquals(restoreString, "rat"); //asserting restored string
     }
 
     @Test
-    public void testRestoreString4()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString2() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedString.setString("invalid"); //setting string
+            int[] array = {1, 0}; //invalid indices array
+            manipulatedString.restoreString(array); //should throw exception
+        });
     }
 
     @Test
-    public void testRestoreString5()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString3() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedString.setString("outofbounds"); //setting string
+            int[] array = {1, 0, 20}; //out of bounds indices array
+            manipulatedString.restoreString(array); //should throw exception
+        });
     }
-
-}
```

Comment:
Everything looks good, however you could have added a couple more test cases for the restoreString method. That's just my personal opinion.

Reply:
Hi @pnarasimhan2021, thank you for the feedback.
It seems I have inadvertently deleted those tests for restore string in my assignment and code review submission.

Will update it for this one. Thank you!

Reply:
@pnarasimhan2021 Now added complete tests.